src/components/table/table-row/table-row.js

```diff
@@ -170,37 +196,37 @@ class TableRow extends React.Component {
   }
 
   /**
-   * @method componentWillUnmount
-   * @return {Void}
-   */
-  componentWillUnmount() {
-    if (this.context.detachFromTable) {
-      this.context.detachFromTable(this.rowID);
-    }
-  }
-
-  /**
    * @method componentWillReceiveProps
    * @return {Void}
    */
   componentWillReceiveProps(nextProps) {
-    if (this.props.uniqueID != nextProps.uniqueID) {
+    if (this.props.uniqueID !== nextProps.uniqueID) {
```

Comment:
Going through each case where `==` is currently used would ideally need to be thoroughly checked to understand the specific use case. As Toni points out, blanket switching to `===` could potentially break the existing behaviour.

I'm not sure there's a single fix that can be applied either. For example, I'm sure in some cases it should only be comparing integers and in others only comparing strings. I suspect this will be quite time consuming to ensure the code is updated correctly.

I'm happy to do this if required, but alternative option would be temporarily remove the rule, get the bulk of the linting changes in and come back to more fiddlier bits like this.